### PR TITLE
roon-{bridge,server}: use stable src urls

### PR DESCRIPTION
**Pull request by lovesegfault**
- roon-server: use upstream stable URL for src
- roon-bridge: use upstream stable URL for src

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/roon-server/default.nix [hunk 1]
```diff
@@ -9,16 +9,18 @@
...
+}:
+let
+  major = "1";
+  minor = "8";
+  build = "831";
+in
+stdenv.mkDerivation {
   pname = "roon-server";
-  version = "1.8-831";
+  version = "${major}.${minor}-${build}";
```
Comment (veprbl) [inline pkgs/servers/roon-server/default.nix:20]: This breaks @r-ryantm . The tarball name could be generated with `builtins.replaceStrings [ "." "-" ] [ "00" "00" ]`

Comment (lovesegfault) [reply pkgs/servers/roon-server/default.nix:20]: Nice catch, fixed :)

**Comment by lovesegfault** [comment]
Result of `nixpkgs-review pr 139664` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>roon-bridge</li>
    <li>roon-server</li>
  </ul>
</details>

**Comment by lovesegfault** [comment]
Result of `nixpkgs-review pr 139664` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>roon-bridge</li>
  </ul>
</details>

**Comment by lovesegfault** [comment]
Result of `nixpkgs-review pr 139664` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>roon-bridge</li>
    <li>roon-server</li>
  </ul>
</details>

File: pkgs/servers/roon-server/default.nix [hunk 2]
```diff
@@ -9,18 +9,19 @@
...
-  src = fetchurl {
-    url = "https://web.archive.org/web/20210921161727/http://download.roonlabs.com/builds/RoonServer_linuxx64.tar.bz2";
-    sha256 = "sha256-SeMSC7K6DV7rVr1w/SqMnLvipoWbypS/gJnSZmpfXZk=";
-  };
+  src =
+    let
+      urlVersion = builtins.replaceStrings [ "." "-" ] [ "00" "00" ] version;
+    in
+    fetchurl {
+      url = "http://download.roonlabs.com/builds/RoonServer_linuxx64_${urlVersion}.tar.bz2";
```
Comment (SuperSandro2000) [inline pkgs/servers/roon-server/default.nix:22]:
```suggestion
  src = fetchurl {
      url = "http://download.roonlabs.com/builds/RoonServer_linuxx64_${lib.replaceStrings [ "." "-" ] version}.tar.bz2";
```

File: pkgs/servers/roon-bridge/default.nix [hunk 3]
```diff
@@ -15,20 +15,20 @@ stdenv.mkDerivation rec {
...
-      sha256 = "sha256-dersaP/8qkl9k81FrgMieB0P4nKmDwjLW5poqKhEn7A=";
-    };
-    aarch64-linux = fetchurl {
-      url = "https://web.archive.org/web/20210803071334/http://download.roonlabs.com/builds/RoonBridge_linuxarmv8.tar.bz2";
-      sha256 = "sha256-zZj7PkLUYYHo3dngqErv1RqynSXi6/D5VPZWfrppX5U=";
-    };
-  }.${system} or throwSystem;
+  src =
+    let
+      urlVersion = builtins.replaceStrings [ "." "-" ] [ "00" "00" ] version;
```
Comment (SuperSandro2000) [inline pkgs/servers/roon-bridge/default.nix:20]:
```suggestion
      urlVersion = builtins.replaceStrings [ "." "-" ] version;
```

File: pkgs/servers/roon-server/default.nix (hunk 2 above)
Comment (lovesegfault) [reply pkgs/servers/roon-server/default.nix:22]: I maintain this alone, I'll keep my style preferences.

Comment (SuperSandro2000) [reply pkgs/servers/roon-server/default.nix:22]: That is nice but replacing twice zero with twice zero does nothing.

**Pull request merged**